src/test/java/io/github/test/classgraph/issues/issue321/ClassInfoListTest.java

```diff
@@ -0,0 +1,35 @@
+package io.github.test.classgraph.issues.issue321;
+
+import static org.assertj.core.api.Assertions.assertThat;
+
+import java.util.Collections;
+import java.util.List;
+import java.util.Optional;
+
+import org.junit.Test;
+
+import io.github.classgraph.ClassGraph;
+import io.github.classgraph.ClassInfo;
+import io.github.classgraph.ClassInfoList;
+import io.github.classgraph.ScanResult;
+
+public class ClassInfoListTest
+{
+
+   @Test
+   public void testNames()
+   {
+      try (ScanResult result = new ClassGraph()
+         .removeTemporaryFilesAfterScan()
```

Comment:
`.removeTemporaryFilesAfterScan()` is not needed in general, whenever you call `ScanResult#close()`, temporary files are deleted. This is only needed if you think the `ScanResult` may not be appropriately closed.